src/pip/_internal/utils/temp_dir.py

```diff
@@ -101,16 +108,18 @@ class TempDirectory(object):
     def __init__(
         self,
         path=None,    # type: Optional[str]
-        delete=None,  # type: Optional[bool]
+        delete=_default,  # type: Union[Optional[bool], _Default]
         kind="temp",  # type: str
         globally_managed=False,  # type: bool
     ):
         super(TempDirectory, self).__init__()
 
         # If we were given an explicit directory, resolve delete option now.
         # Otherwise we wait until cleanup and see what tempdir_registry says.
-        if path is not None and delete is None:
+        if path is not None and delete is _default:
             delete = False
+        elif delete is _default:
+            delete = None
```

Comment:
I needed to pause a few seconds to realise `__exit__` does not need additional handling because `_default` is truthy. Is there a way to improve readability in this regard?

Reply:
You mean it was not clear that `self.delete` would never be `_delete`? If so I can nest both conditions under a `if delete is _default` and make it clear that `delete` is always reset.

Reply:
Yeah, that’d be more obvious 👍

Reply:
Just had another thought; maybe it’d be enough to explicit mark `self.delete` as `Optional[bool]`, that would quench any doubt `_default` could carry over into `__exit__`. Not sure how best to do this with type comments though.

Reply:
It ended up looking pretty readable after factoring under a single condition [here](https://github.com/pypa/pip/pull/7652/files#diff-bd9cdc0022571f341d8a451e918e279bR117-R125).